### PR TITLE
[GOV] CC and CoC term limitation to 2 years

### PR DESCRIPTION
**Pull request by aiwalter**
#### Reference Issues/PRs
#3865 

#### What does this implement/fix? Explain your changes.
Clarify tenure ending CC and CoC

### CODE REVIEW

**Comment by fkiraly** [comment]
Formally voting this down.

I want to see requirements to contribute to the community in some ways if we do this.

E.g., what is the check of a dogpile of people voting themselves in and plundering sktime?
Hypothetically, 5 people who don't really do anything to maintain sktime, don't code, don't mentor, don't help with events, or are not welcoming to new contributors.

**Comment by fkiraly** [comment]
If we do this, there needs to be a "proper" election system, or we may end up with a single remaining CC member when core devs cannot agree on a supermajority for a time.

**Comment by aiwalter** [comment]
I cant see how a missing number of volunteers for a position is related to a voting system?  Core developers can always have a vote to end tenure of a person at any time, e.g. if there is only 1 person at CC or CoC. But if you want we could add a phrase, e.g. "Minimum number for this committee to be active are two persons, if there is only one person, the committee has to be inactive until at least two persons are member again."

**Comment by fkiraly** [comment]
> "Minimum number for this committee to be active are two persons, if there is only one person, the committee has to be inactive until at least two persons are member again."

Sounds like a bad idea if you want to ensure the lights are being kept on.

In a situation where the core dev community is 50/50 divided, or 33/33/33 divided, this leads to the CC "phasing out" and sktime governance moving into anarchy.

**Comment by aiwalter** [comment]
Feel free to propose sth. I think this PR is an improvement towards democratization of GOV, you can ofc propose additions to it in a separate PR

**Comment by fkiraly** [comment]
> Feel free to propose sth.

That proposal already exists, and you should be referencing it and discussing it as a proper alternative.

https://github.com/sktime/sktime/pull/3697

ANother alternative: if the vote is split (no 2/3 majority either way), CC member stays by default.

**Comment by GuzalBulatova** [comment]
I'd vote this: if there's no 2/3 "for" having the CC member for the "second term" then they're not there for the second term. To avoid anarchy we can have this option to leave the CC consisting of the old members until there are new ones elected by majority.

**Review by GuzalBulatova — CHANGES_REQUESTED** [review]
@aiwalter can you please add the description for cases where there has been no new members of CC elected and old members haven't been re-elected for the second term? My proposal is to have the "old CC" effective until the new members have been re-elected, min 2, preferably 3 in total

**Comment by aiwalter** [comment]
> can you please add the description for cases where there has been no new members of CC elected and old members haven't been re-elected for the second term? 

In my opinion this is actually a different construction side. We had this scenarios in the past at CoCC when people resigned and now its only one person. So having such scenarios is independent to this PR in my opinion. We can have a separate PR with sth like I proposed above? "Minimum number for this committee to be active are two persons, if there is only one person, the committee has to be inactive until at least two persons are member again." That would then also catch your proposal:

> My proposal is to have the "old CC" effective until the new members have been re-elected, min 2, preferably 3 in total

I think its better tho have CC or CoCC inactive than having someone there who does not have 2/3 of core dev votes. So @GuzalBulatova  do you want to make a separate PR for min required members of CC and CoCC?

I feel discussion about min numbers for a committee is just an alibi discussion to not get this PR in? I wanted no purpose to keep the gov PRs I made with minimal content each so we can streamline the discussion and focus on a particualar gov item. In this case the tenure ending. 

@fkiraly 

> Sounds like a bad idea if you want to ensure the lights are being kept on.

Yesterday you mentioned in Slack the idea to abolish CC, now you think its a bad idea to have CC inactive for temporary time if required to find a new committe member. Is this not a it contradictory?

**Review by MatthewMiddlehurst — APPROVED** [review]
I agree, having a temporarily inactive CC is more suitable than keeping on members which the core-developers do not want to represent them.

**Review by TonyBagnall — APPROVED** [review]

**Review by lmmentel — DISMISSED** [review]

File: docs/source/get_involved/governance.rst [hunk 1]
```diff
@@ -301,6 +301,14 @@ anonymous.
 To avoid deadlocks if there is an even number of CoC committee members, one
 of them will have a tie breaking privilege.
 
+End of tenure
+^^^^^^^^^^^^^
+
+CoC members can resign voluntarily at any point in time, by informing the CoC or core developers in writing.
+
+CoC membership ends automatically after a person has been in this role for 2 years. If the person would like to continue the membership of CoC, a new nomination vote has to happen following the procedure as for nomination of a new CoC member.
+The vote can be requested by the affected CoC member or by any core developer.
```
Comment (aiwalter) [inline docs/source/get_involved/governance.rst:310]:
```suggestion
CoC membership ends automatically after a person has been in this role for 2 years. If the person would like to continue the membership of CoC, a new appointment vote has to happen following the procedure as for appointment of a new CoC member.
The vote can be requested by the affected CoC member or by any core developer.
```

File: docs/source/get_involved/governance.rst [hunk 2]
```diff
@@ -367,14 +375,17 @@ and need not follow their anonymous vote if they have voted.
...
+CC members can resign voluntarily at any point in time, by informing the CC or core developers in writing.
 
 CC members who do not actively engage with their CC member role responsibilities are
 expected to resign voluntarily.
 
 Tenure also ends automatically when a CC member's tenure as core developer ends,
 e.g., through resignation or inactivity.
 
+CC membership ends automatically after a person has been in this role for 2 years. If the person would like to continue the membership of CC, a new nomination vote has to happen following the procedure as for nomination of a new CC member.
+The vote can be requested by the affected CC member or by any core developer.
```
Comment (aiwalter) [inline docs/source/get_involved/governance.rst:387]:
```suggestion
CC membership ends automatically after a person has been in this role for 2 years. If the person would like to continue the membership of CC, a new appointment vote has to happen following the procedure as for appointment of a new CC member.
The vote can be requested by the affected CC member or by any core developer.
```

**Comment by fkiraly** [comment]
formal downvote.

Reasons explained here: https://github.com/sktime/sktime/issues/3882

**Review by GuzalBulatova — DISMISSED** [review]
There was no intent from my side to keep this PR open, I was expressing concerns regarding special cases, @aiwalter . But agree that it should then just be in a separate PR, with the aim of keeping the changes proposed minimal.

**Comment by aiwalter** [comment]
a core dev vote was invoked on this PR in the private core dev Slack channel and will end after 7 days.

**Review by lmmentel — APPROVED** [review]

**Review by ltsaprounis — APPROVED** [review]

**Comment by fkiraly** [comment]
This PR has violated governance rules on how to vote multiple times.

* gov rules clearly say, we vote publicly. Martin opened votes in private.
* Martih has used some app to vote anonymously in a private slack channel that had people in it who were not eligible to vote.
* this is a substantial change so would formally require a STEP, which implies 7 days discussion period.

Further, we are violating the "spirit" of how to vote.

* some people are now on holiday and will be unable to vote until they return.
* some people wanted to pause from votes after this and will not see the second vote in time.

I think we should just stop this and base the changes on consensus.

I.e., start over in January when everyone is back from holidays or pause.
Then, work out how elections work with term limits and clear rules, and have NumFOCUS arbitrate (hopefully we get accepted) in cases of disagreement.

**Review by patrickzib — APPROVED** [review]

**Pull request closed without merging**